### PR TITLE
Skip failing tests in WP 5.3 and 5.4

### LINKED ISSUE #6194: Incompatibility with Wordpress 5.3 and 5.4 (not present in 5.5+)
While building the new unit test matrix in #6164 I noticed a consistent failure in matrix runs of Wordpress 5.3 and 5.4.

The stack trace for the test failure:

```
3) WC_Tests_API_Reports_Orders::test_custom_product_attributes_filter
Undefined index: items
/tmp/wordpress/wp-includes/rest-api.php:1237
/tmp/wordpress/wp-includes/rest-api.php:1237
/tmp/wordpress/wp-includes/rest-api.php:1037
/tmp/wordpress/wp-includes/rest-api/class-wp-rest-request.php:885
/tmp/wordpress/wp-includes/rest-api/class-wp-rest-server.php:884
/tmp/wordpress/wp-content/plugins/woocommerce/tests/legacy/framework/vendor/class-wp-test-spy-rest-server.php:35
/tmp/wordpress/wp-content/plugins/woocommerce-admin/tests/api/reports-orders.php:303
```

It seems this is a real incompatibility that needs investigating and fixing. But first we need to reproduce it in a local dev site. To move forward the unit test migration we've opted to leave 5.3 and 5.4 out of our test runs until we can fix this.

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #6194 

This PR skips the tests that fail only in WP 5.3 and 5.4 due to the difference in how rest-api.php handles schema validation.

More details on here https://github.com/woocommerce/woocommerce-admin/issues/6194#issuecomment-768052452

### Detailed test instructions:

1. Remove the existing phpunit volume `docker volume rm -f wc-admin-php-test-suite_test-suite`
2. Open docker/wc-admin-php-test-suite/entrypoint.sh
2. Change the following line.

Before: `	bin/install-wp-tests.sh $DB_NAME $DB_USER $DB_PASS $DB_HOST latest true`
After: `	bin/install-wp-tests.sh $DB_NAME $DB_USER $DB_PASS $DB_HOST 5.3 true`

3. Run the test normally and confirm three tests are skipped.

### CODE REVIEW

File: tests/api/reports-orders.php [hunk 1]
```diff
@@ -236,6 +241,11 @@ public function test_product_attributes_filter() {
 	 * Test filtering by custom product attribute(s).
 	 */
```
Comment (adrianduffell) [inline tests/api/reports-orders.php:242]:
```suggestion
	 *
	 * Skipped in WP 5.3 & 5.4 due to a bug in WordPress when validating arrays.
	 */
```

File: tests/api/reports-orders.php [hunk 2]
```diff
@@ -133,6 +133,11 @@ public function test_reports_schema() {
 	 * Test filtering by taxonomy-backed product attribute(s).
 	 */
```
Comment (adrianduffell) [inline tests/api/reports-orders.php:134]:
```suggestion
	 *
	 * Skipped in WP 5.3 & 5.4 due to a bug in WordPress when validating arrays.
	 */
```

File: tests/api/reports-orders-stats.php [hunk 3]
```diff
@@ -132,6 +132,11 @@ public function test_reports_schema() {
 	 * Test filtering by product attribute(s).
 	 */
```
Comment (adrianduffell) [inline tests/api/reports-orders-stats.php:133]:
```suggestion
	 *
	 * Skipped in WP 5.3 & 5.4 due to a bug in WordPress when validating arrays.
	 */
```

**Review by adrianduffell — APPROVED** [review]
Nice investigation on this issue @moon0326 👍 

Code looks good and the tests pass for me. Could we add some documentation about why the tests are skipped? I left some suggestions you could commit. Then I think it's ready to merge.

File: tests/api/reports-orders.php (hunk 1 above)
Comment (moon0326) [reply tests/api/reports-orders.php:242]: Thank you for the suggestion @adrianduffell 👍 I will commit that.

Comment (adrianduffell) [reply tests/api/reports-orders.php:242]: That's perfect, thanks!

**Pull request merged**